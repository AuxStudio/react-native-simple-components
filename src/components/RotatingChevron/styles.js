import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
    transform: [{ rotate: '270deg' }],
    textAlign: 'center',
  },
});

export default styles;
