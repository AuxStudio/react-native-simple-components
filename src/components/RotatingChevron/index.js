import React from 'react';
import { ViewPropTypes, Text } from 'react-native';
import PropTypes from 'prop-types';
import { AnimateRotate } from 'react-native-simple-animators';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import Touchable from '../Touchable';

export default class RotatingChevron extends React.Component {
  constructor(props) {
    super(props);

    this.toggleRotate = this.toggleRotate.bind(this);

    this.state = {
      shouldRotate: false,
    };
  }

  static get propTypes() {
    return {
      handlePress: PropTypes.func,
      androidRipple: PropTypes.bool,
      androidRippleColor: PropTypes.string,
      androidRippleBorderless: PropTypes.bool,
      iconStyle: Text.propTypes.style,
      style: ViewPropTypes.style,
      wrapperStyle: ViewPropTypes.style,
    };
  }

  static defaultProps = {};

  toggleRotate(event) {
    this.setState({
      shouldRotate: !this.state.shouldRotate,
    });

    if (this.props.handlePress) {
      this.props.handlePress(event);
    }
  }

  render() {
    return (
      <AnimateRotate
        initialValue={0}
        finalValue={-180}
        shouldAnimateIn={this.state.shouldRotate}
        shouldAnimateOut={!this.state.shouldRotate}
        style={this.props.wrapperStyle}
      >
        <Touchable
          onPress={this.toggleRotate}
          androidRipple={this.props.androidRipple}
          androidRippleColor={this.props.androidRippleColor}
          androidRippleBorderless={this.props.androidRippleBorderless}
          style={this.props.style}
        >
          <Icon name="chevron-left" style={[styles.icon, this.props.iconStyle]} />
        </Touchable>
      </AnimateRotate>
    );
  }
}
