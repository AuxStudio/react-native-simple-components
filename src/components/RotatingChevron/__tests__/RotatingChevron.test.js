import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-native-simple-animators', () => ({
  AnimateRotate: 'AnimateRotate',
}));

import RotatingChevron from '../'; // eslint-disable-line

describe('RotatingChevron', () => {
  it('renders with all props', () => {
    const component = renderer.create(
      <RotatingChevron
        handlePress={jest.fn()}
        androidRipple
        androidRippleColor="red"
        androidRippleBorderless
        showShadow
        iconStyle={{ color: 'red' }}
        style={{ backgroundColor: 'blue' }}
        wrapperStyle={{ backgroundColor: 'green' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<RotatingChevron />)).toMatchSnapshot();
  });

  // it('renders and toggles rotate', () => {
  //   const component = renderer.create(
  //     <RotatingChevron
  //       handlePress={jest.fn()}
  //       androidRipple
  //       androidRippleColor="red"
  //       androidRippleBorderless
  //       showShadow
  //       iconStyle={{ color: 'red' }}
  //       style={{ backgroundColor: 'blue' }}
  //       wrapperStyle={{ backgroundColor: 'green' }}
  //     />,
  //   );
  //   const instance = component.getInstance();
  //   instance.toggleRotate();
  //   expect(instance.state.shouldRotate).toBe(true);
  // });
});
