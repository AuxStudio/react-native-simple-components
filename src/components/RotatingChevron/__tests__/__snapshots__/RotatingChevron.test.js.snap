// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotatingChevron renders with all props 1`] = `
<AnimateRotate
  finalValue={-180}
  initialValue={0}
  shouldAnimateIn={false}
  shouldAnimateOut={true}
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#212121",
              "fontSize": 24,
              "textAlign": "center",
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {
              "color": "red",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</AnimateRotate>
`;

exports[`RotatingChevron renders with minimum required props 1`] = `
<AnimateRotate
  finalValue={-180}
  initialValue={0}
  shouldAnimateIn={false}
  shouldAnimateOut={true}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#212121",
              "fontSize": 24,
              "textAlign": "center",
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</AnimateRotate>
`;
